lib/msal-common/src/url/UrlString.ts

```diff
@@ -145,15 +140,34 @@ export class UrlString {
         return urlComponents;
     }
 
+    /**
+     * Parses hash string from given string. Returns empty if no string is given.
+     * @param hashString 
+     */
+    static parseHash(hashString: string): string {
+        const hashIndex1 = hashString.indexOf("#");
+        const hashIndex2 = hashString.indexOf("#/");
+        if (hashIndex2 > -1) {
+            return hashString.substring(hashIndex2 + 2);
+        } else if (hashIndex1 > -1) {
+            return hashString.substring(hashIndex1 + 1);
+        }
+        return "";
+    }
+
     static constructAuthorityUriFromObject(urlObject: IUri): UrlString {
         return new UrlString(urlObject.Protocol + "//" + urlObject.HostNameAndPort + "/" + urlObject.PathSegments.join("/"));
     }
 
     /**
-     * Returns deserialized portion of URL hash
+     * Returns URL hash as server auth code response object.
      */
     static getDeserializedHash(hash: string): ServerAuthorizationCodeResponse {
```

Comment:
Would "urlString" be a more apt name instead of "hash"?

Reply:
no because this function specifically takes a hash string. we never pass the entire url here

Reply:
In that case let's update `hashContainsKnownProperties` to be consistent with that. 

Also, why are we then checking if the hash is empty on both line 166 and 170?

Reply:
> In that case let's update hashContainsKnownProperties to be consistent with that.

`hashContainsKnownProperties` does have use cases where the value passed to it is a full url in `@azure/msal-browser`. We can move it to that package, but I figured it would be fine to leave this here since it is a protocol level API.

> Also, why are we then checking if the hash is empty on both line 166 and 170?

It's a bit of a weird scenario, but line 166 checks the value that is passed in. line 170 checks that the deserialized hash is parsed correctly. That's why one scenario throws an error and one returns an empty string. I will add comments.

Reply:
The way I read this is: 
Url passed in -> !Url.isEmpty -> Attempt to extract the hash from the url -> Attempt to deserialize the hash

I would maybe try to rename some things because `parseHash` is really just taking the right side of the anchor, it's not parsing anything. 

I'd also suggest maybe returning the passed "hash" in `parseHash` if there is no anchor tag, rather than having 2 `isEmpty` calls adding to the confusion

Reply:
resolved offline.